### DISCUSSION TITLE
How can I add custom key-value metadata to the PDF document?

### DISCUSSION THREAD

**Discussion by ramkumarkb** [typst/typst#4473] [Ideas]
This request / question could be related to the issue - https://github.com/typst/typst/issues/2266 

PDF Specifications seems to allow this and is mentioned in the page-548 of the v1.7 of PDF (metadata stream)

From my read thus far, the only way we can do this is Typst today is via `keywords` string and add something like `key1=value1|key2=value2` etc - https://typst.app/docs/reference/model/document/#parameters-keywords 

Ideally, one can add custom key/value as custom document properties

**Comment by TS60** [comment]
Currently, this is not possible from within Typst.
You have two options to achieve this:
1. Use a post-processor to edit the PDF file and set the metadata there. This does not always keep the PDF file as is, some post-processors remove links, attachements and other things.
2. Reserve space in the Metadata for example in the mentioned `keywords` you could reserve a string of some big length like `"aaaaaa..."`
    After this, you have enough memory in the Metadata dict to rewrite it yourself using a python script for example. It is important that afterwards, the metadata dict has the same length like before and starts and ends at the same byte position. If the metadata dict is shorter than the reserved space, the space should be filled up with spaces.
